### DISCUSSION TITLE
How do I call LSP workspace command that accepts parameters?

### DISCUSSION THREAD

**Discussion by Esgariot** [helix-editor/helix#9210] [Q&A]
I'm trying to perform a typescript file rename, such that all the imports would get updated.

I see there's [`_typescript.applyRenameFile`](https://github.com/typescript-language-server/typescript-language-server/blob/master/README.md#rename-file), which I'd like to use, (I'm aware of [`:move`/`willRename`](https://github.com/helix-editor/helix/pull/8584) but it has not been released, as of writing.)

Looking at https://github.com/helix-editor/helix/blob/master/helix-term/src/ui/mod.rs#L351 I also see there's completer for workspace commands, but I can't get it to accept further positional arguments, it treats whole expression as a command name.

trying to run command as-is:
```
2024-01-02T18:18:36.623 helix_term::commands::lsp [ERROR] execute LSP command: protocol error: InternalError: Request workspace/executeCommand failed with message: Cannot destructure property 'sourceUri' of 'e.arguments[0]' as it is undefined.
```

What am I missing? Any workarounds?

**Comment by Feel-ix-343** [comment]
related: https://github.com/Feel-ix-343/markdown-oxide/issues/115

**Comment by initshdb** [comment]
Just figured out that `:lsp-workspace-command` lets you call LSP workspace commands.

@Feel-ix-343 For markdown-oxide, I could not see the `Daily` command, however the Predefined commands work fine inside Helix - `:lsp-workspace-command today` takes me to today's note, creates if it does not exit :)
